### ISSUE TITLE
Add Tilezen vector tile support

### ISSUE THREAD

**Issue by nvkelso** [geometalab/Vector-Tiles-Reader-QGIS-Plugin#78]
Really nice to see this QGIS plugin develop, great work! 💯 

I'm trying to add support for Mapzen vector tiles but hitting a confusing error state.

My TileJSON is here: 

- http://tile.mapzen.com/mapzen/vector/v1/tilejson/mapbox.json?api_key=mapzen-zD4v4XB

That correctly loads in the interface:

![screen shot 2017-09-12 at 15 17 15](https://user-images.githubusercontent.com/853051/30350961-f59479a8-97cd-11e7-9ccb-c1d0a5ad017d.png)

But clicking the Add button results in nothing being added. The following is from the Python console (inside QGIS):

```
Bounds of source: [-124.23126252746184, 37.53840571136699, -120.63988204844593, 38.04268178855649]
Current extent: {'y_min': 789, 'y_max': 793, 'height': 5, 'width': 21, 'x_max': 337, 'x_min': 317}
Load: http://tile.mapzen.com/mapzen/vector/v1/tilejson/mapbox.json?api_key=mapzen-zD4v4XB
Calculating 105 tiles
0 cache hits. 105 may potentially be loaded.
Unexpected error during logging: tuple index out of range
Original message: 'Loading extent {} for zoom level '{}' of: {}', params: '(11, u'mapzen mapbox format')'
Loading complete! Loaded extent: None
```

While a good load from OpenMapTiles is:

```
Bounds of source: [-124.23126252746184, 37.53840571136699, -120.63988204844593, 38.04268178855649]
Current extent: {'y_min': 789, 'y_max': 793, 'height': 5, 'width': 21, 'x_max': 337, 'x_min': 317}
Bounds of source: [-180, -85.0511, 180, 85.0511]
Load: 
Calculating 105 tiles
105 cache hits. 0 may potentially be loaded.
Unexpected error during logging: tuple index out of range
Original message: 'Loading extent {} for zoom level '{}' of: {}', params: '(11, u'OpenMapTiles')'
Creating hierarchy in QGIS
Loading complete! Loaded extent: {'x_min': 317, 'y_max': 793, 'y_min': 789, 'x_max': 337}
```

Notably it's the final 2 lines that are different:

```
Creating hierarchy in QGIS
Loading complete! Loaded extent: {'x_min': 317, 'y_max': 793, 'y_min': 789, 'x_max': 337}
```

Any tips on the above error? 

Any suggestions for refactoring the plugin to support different default styles per "connection" tile schema?

**Comment by sfkeller** [comment]
Hi Nathaniel, Thanks for testing and contributing. I'm very enthusiastic about this plugin (too). 

Unfortunately this plugin was meant as a prototype and is/was a semester student project which actually came to an end these days. (We're struggling currently to get thet last great issue (#73) done, which would finally give a proof of concept that Vector Tiles are working as base map even inside a GIS.)

So I'd like to keep expectations low about how active we can maintain this plugin/repo.

**Comment by sfkeller** [comment]
> Any suggestions for refactoring the plugin to support different default styles per "connection" tile schema?

I'd like to put that in a separate issue (see #80).

**Comment by nvkelso** [comment]
Cool, I've added some details to #80. 

So far the proof of concept is very good and usable as a v1.0 launch I think. I'd like to promote it at the upcoming NACIS conference (and would prefer to also add Mapzen support by then).

**Comment by mnboos** [comment]
Mapzen now also works with the plugin. There were two issues where I created the url when loading a tile.

Notice: if a parameter "api_key" is available in the url of the source and a value "{api_key}" is available in the url of any tile, it will be replaced by the api key of the source.

**Comment by mnboos** [comment]
Version 1.0.2 of the plugin will be published soon.

**Comment by nvkelso** [comment]
Awesome! What about default styling, too? Is that something I can help with? (Since the tile Schema is different.)

**Comment by nvkelso** [comment]
Confirmed the new version is updated to QGIS plugin listing and I can select the new Mapzen option in the dialog and load tile content into QGIS (tested in San Francisco Bay Area)!

<img width="625" alt="screen shot 2017-09-22 at 15 22 06" src="https://user-images.githubusercontent.com/853051/30766427-d5c0d624-9fa9-11e7-92d9-510f3aee042f.png">

<img width="225" alt="screen shot 2017-09-22 at 15 22 10" src="https://user-images.githubusercontent.com/853051/30766428-d5c34eae-9fa9-11e7-8a9a-8b9d888b6623.png">

**Comment by sfkeller** [comment]
Due to the shutdown of Mapzen services end of January 2018 we unfortunately have to remove this from predefined online sources of the QGIS plugin.
 
Perhaps someone steps in and takes over the publishing of these nice vector tile service?

**Comment by nvkelso** [comment]
There is some movement that direction, please don’t remove it just yet... and the current Mapzen hostess Tilezen tiles will work thru Jan 2018.

I’m the end hopefully it’s just a config change to new URL endpoint.

**Comment by sfkeller** [comment]
@nvkelso: Great news: https://mapzen.com/blog/long-term-support-mapzen-maps/ . Eager to receive instructions from you cool guys (cheers also to Ian and Matt)!

**Comment by nvkelso** [comment]
Hello! You're certainly on it :) We're still working out some final
details, be in touch.

**Comment by sfkeller** [comment]
@nvkelso: Any update on Mapzen tiles (and to #101  )? We're going to integrate Mapcat and I'd love to keep Mapzen remaining as another VT source!

**Comment by mnboos** [comment]
The Mapzen.com connection isn't working anymore and due to this, I'll remove it from the plugin.

**Comment by nvkelso** [comment]
@mnboos Is there a blocker to using Tilezen vector tiles via Nextzen endpoint?

**Comment by nvkelso** [comment]
Once https://github.com/tilezen/tapalcatl-py/issues/23 is completed the Nextzen tiles will add support for tilejson, which I think unblocks that use in this plugin?

**Comment by mnboos** [comment]
@nvkelso Is there a replacement api-key that can be used for the nextzen connection? The mapzen key isn't working on nextzen, unsurprisingly.

**Comment by nvkelso** [comment]
Details are at:

https://developers.nextzen.org/

**Comment by mnboos** [comment]
@sfkeller Nextzen doesn't seem to support TileJSON, at least it's not listed under https://developers.nextzen.org/about.html. This means, we cannot provide a nextzen predefined connection at the moment.

**Comment by nvkelso** [comment]
It does, we’ll add documentation for that.

**Comment by mnboos** [comment]
You wrote:

> Details are at:
> https://developers.nextzen.org/

I couldn't find anything there, maybe I just didn't see it. As soon as I know how to handle nextzen, I'll readd the connection.

**Comment by nvkelso** [comment]
Server needed to be tickled, it’s up now:

https://tile.nextzen.org/tilezen/vector/v1/512/all/tilejson.mvt.json?api_key=your-nextzen-api-key

**Comment by mnboos** [comment]
@nvkelso May we activate the nextzen predefined connection within the plugin?

**Comment by nvkelso** [comment]
Of course!

**Comment by mnboos** [comment]
Cool thanks!

I noticed there is a server placeholder in the address (`{s}`) and defined the servers [a-d]. Is this okay or would you recommend something else?

**Comment by nvkelso** [comment]
Correct, that endpoint supports [a-d] for {s}.
